### PR TITLE
ARROW-4005: [Plasma] [GLib] Add gplasma_client_disconnect()

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/plasma-glib/client.cpp [hunk 1]
```diff
@@ -431,6 +431,24 @@ gplasma_client_refer_object(GPlasmaClient *client,
...
+ * Returns: %TRUE on success, %FALSE if there was an error.
+ *
+ * Since: 0.12.0
+ */
+gboolean
+gplasma_client_disconnect(GPlasmaClient *client,
+                          GError **error)
+{
+  auto plasma_client = gplasma_client_get_raw(client);
+  auto status = plasma_client->Disconnect();
```
Comment (kou) [inline c_glib/plasma-glib/client.cpp]: We need to keep "disconnected" information.
If users don't call `gplasma_client_disconnect()` explicitly, we call `Disconnect()` on `fainalize()`.

```cpp
if (garrow_error_check(...)) {
  priv->disconnected = true;
  return TRUE;
} else {
  return FALSE;
}
```

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Can you check my comment?

File: c_glib/plasma-glib/client.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/plasma-glib/client.cpp]: I’ll keep "disconnected” information. If `priv->disconnected` is true, I will not call `Disconnected` on finalize().

File: c_glib/plasma-glib/client.cpp [hunk 2]
```diff
@@ -431,6 +434,29 @@ gplasma_client_refer_object(GPlasmaClient *client,
...
+ * Since: 0.12.0
+ */
+gboolean
+gplasma_client_disconnect(GPlasmaClient *client,
+                          GError **error)
+{
+  auto priv = GPLASMA_CLIENT_GET_PRIVATE(client);
+  auto status = priv->client->Disconnect();
+  if (garrow_error_check(error, status, "[plasma][client][disconnect]")) {
+    priv->disconnected = true;
```
Comment (kou) [inline c_glib/plasma-glib/client.cpp]: We should use `bool` instead of `gboolean` when we use `true`.

Comment (shiro615) [reply c_glib/plasma-glib/client.cpp]: I see. I'll use `bool`.

File: c_glib/test/plasma/test-plasma-client.rb [hunk 3]
```diff
@@ -84,4 +84,13 @@ def setup
                    ])
     end
   end
+
+  test("#disconnect") do
+    require_gi(1, 42, 0)
```
Comment (kou) [inline c_glib/test/plasma/test-plasma-client.rb]: Can we remove this by using options?

**Review by kou — CHANGES_REQUESTED** [review]

File: c_glib/test/plasma/test-plasma-client.rb (hunk 3 above)
Comment (shiro615) [reply c_glib/test/plasma/test-plasma-client.rb]: Yes. I've removed `require_gi`.

**Review by kou — APPROVED** [review]
+1
I'll merge this when CI is green.

**Pull request closed without merging**